src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp

```diff
@@ -635,8 +639,8 @@ void CachedOnDiskReadBufferFromFile::predownload(FileSegmentPtr & file_segment)
                 /// TODO: allow seek more than once with seek avoiding.
 
                 bytes_to_predownload = 0;
-                file_segment->completeWithState(FileSegment::State::PARTIALLY_DOWNLOADED_NO_CONTINUATION);
 
+                chassert(file_segment->state() == FileSegment::State::PARTIALLY_DOWNLOADED_NO_CONTINUATION);
```

Comment:
Where this state expected to be set if we entered this block after 

```cpp
LOG_TEST(log, "Bypassing cache because writeCache (in predownload) method failed");
continue_predownload = false
```

above?

Reply:
Here https://github.com/ClickHouse/ClickHouse/blob/4b9eff7dab23e729d3b68f9dbb1fae42cadbce34/src/Interpreters/Cache/FileSegment.cpp#L503
(Not clearly visible I agree, I actually rewrote this code in a more explicit way here https://github.com/ClickHouse/ClickHouse/pull/44985)